### PR TITLE
KEYCLOAK-12249 added test to test that time is localized

### PR DESCRIPTION
**Pull request by edewit**

### CODE REVIEW

**Comment by edewit** [comment]
UI build: https://keycloak-jenkins.com/job/keycloak-account2-matrix/958/

**Review by ssilvert — COMMENTED** [review]
LGTM.  Test is now passing in Jenkins.

@vmuzikar If you agree I'll merge it.

**Comment by ssilvert** [comment]
> LGTM. Test is now passing in Jenkins.
> 
> @vmuzikar If you agree I'll merge it.

@vmuzikar I thought Ed had already asked you to review.  If you don't have time, that's OK.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/WelcomeScreen.java [hunk 1]
```diff
@@ -41,21 +41,21 @@
     @FindBy(xpath = "//*[@id='" + ROOT_ELEMENT_ID + "']//header")
     private WelcomeScreenHeader header;
 
-    @FindBy(xpath = "//*[@id='landing-personal-info']/a")
+    @FindBy(xpath = "//a[@id='landing-personal-info']")
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/WelcomeScreen.java:44]: Just out of curiosity, what's the reason to make this change?

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java [hunk 2]
```diff
@@ -46,12 +49,10 @@
 import static org.hamcrest.Matchers.hasProperty;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java:52]: Please don't use star imports. It's somewhere specified in our code style but can't find it right now. Correct, @ssilvert?

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java [hunk 3]
```diff
@@ -229,6 +231,34 @@ public void timesTests() {
...
 
+    @Test
+    public void timeLocaleTest() {
+        String sessionId = createSession(Browsers.CHROME);
+        testingClient.server().run(session -> {
+            RealmModel realm = session.realms().getRealmByName(TEST);
+            UserSessionModel userSession = session.sessions().getUserSession(realm, sessionId);
+
+            userSession.setLastSessionRefresh(Time.currentTime());
+        });
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/DeviceActivityTest.java:242]: This is not needed. We don't care about exact time of the session.

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/InternationalizationTest.java [hunk 4]
```diff
@@ -93,6 +105,34 @@ public void userAttributeTest() {
...
     }
 
+    @Test
+    public void shouldDisplayTimeUsingSelectedLocale() {
+        signingInPage.navigateTo();
+        loginToAccount();
+        SigningInPage.UserCredential passwordCred =
+                passwordCredentialType.getUserCredential(testUserResource().credentials().get(0).getId());
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMMM d, yyyy, h:mm a");
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/InternationalizationTest.java:115]: Locale is not specified in the formatter. Is that intentional? I know it defaults to English in most cases but not sure what happens if someone runs the test on a machine with different locale.

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Thanks @edewit! In general looks good, just a few minor comments/questions.

Also, I've verified the German locale works fine in tests with RH-SSO (which normally doesn't include German).

File: testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/WelcomeScreen.java (hunk 1 above)
Comment (edewit) [reply testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/WelcomeScreen.java:44]: I'm also fixing a test in this PR

**Review by vmuzikar — APPROVED** [review]
@edewit Thank you!

**Review by ssilvert — APPROVED** [review]

**Pull request merged**